src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -385,24 +358,33 @@ private void WriteAsPaxSharedInternal(Span<byte> buffer, out long actualLength)
             _checksum = WriteChecksum(tmpChecksum, buffer);
         }
 
-        // All formats save in the name byte array only the ASCII bytes that fit. The full string is returned in the out byte array.
-        private int WriteName(Span<byte> buffer, out byte[] fullNameBytes)
+        // All formats save in the name byte array only the ASCII bytes that fit.
+        private int WriteName(Span<byte> buffer)
         {
-            fullNameBytes = Encoding.ASCII.GetBytes(_name);
-            int nameBytesLength = Math.Min(fullNameBytes.Length, FieldLengths.Name);
-            int checksum = WriteLeftAlignedBytesAndGetChecksum(fullNameBytes.AsSpan(0, nameBytesLength), buffer.Slice(FieldLocations.Name, FieldLengths.Name));
-            return checksum;
+            ReadOnlySpan<char> src = _name.AsSpan(0, Math.Min(_name.Length, FieldLengths.Name));
+            Span<byte> dest = buffer.Slice(FieldLocations.Name, FieldLengths.Name);
+            int encoded = Encoding.ASCII.GetBytes(src, dest);
+            return Checksum(dest.Slice(0, encoded));
         }
 
         // Ustar and PAX save in the name byte array only the ASCII bytes that fit, and the rest of that string is saved in the prefix field.
         private int WritePosixName(Span<byte> buffer)
         {
-            int checksum = WriteName(buffer, out byte[] fullNameBytes);
-            if (fullNameBytes.Length > FieldLengths.Name)
+            int checksum = WriteName(buffer);
+
+            if (_name.Length > FieldLengths.Name)
             {
-                int prefixBytesLength = Math.Min(fullNameBytes.Length - FieldLengths.Name, FieldLengths.Name);
-                checksum += WriteLeftAlignedBytesAndGetChecksum(fullNameBytes.AsSpan(FieldLengths.Name, prefixBytesLength), buffer.Slice(FieldLocations.Prefix, FieldLengths.Prefix));
+                int prefixBytesLength = Math.Min(_name.Length - FieldLengths.Name, FieldLengths.Name);
+                Span<byte> remaining = prefixBytesLength <= 256 ?
+                    stackalloc byte[prefixBytesLength] :
+                    new byte[prefixBytesLength];
```

Comment:
not renting -- name assumed to be never that long? or just because this is rarely called?

Reply:
yes

Reply:
also,  the output is limited to storing FieldLengths.Prefix. If it turned out to be common,  we'd instead want to just limit it to that rather than using the pool.